tensorflow/contrib/go/graph_test.go

```diff
@@ -0,0 +1,252 @@
+package tensorflow_test
+
+import (
+	"testing"
+
+	tf "github.com/tensorflow/tensorflow/tensorflow/contrib/go"
+)
+
+func TestGraphGeneration(t *testing.T) {
+	graph := tf.NewGraph()
+	input1 := graph.Placeholder("input1", tf.DtInt32, []int64{3}, []string{})
+	input2 := graph.Placeholder("input2", tf.DtInt32, []int64{3}, []string{})
+	_, err := graph.Op("Add", "output", []*tf.GraphNode{input1, input2}, "", nil)
+	if err != nil {
+		t.Error("Problem trying add two tensord, Error:", err)
+		t.FailNow()
+	}
+
+	_, err = graph.Op("Add", "output", []*tf.GraphNode{input2}, "", map[string]interface{}{
+		"T": tf.DtInt32,
+	})
+	if err == nil {
+		t.Error("An with two mandatory parameters was added after specify just one")
+	}
+	_, err = graph.Op("Aajajajajdd", "output", []*tf.GraphNode{input2}, "", map[string]interface{}{})
+	if err == nil {
+		t.Error("An undefined operation was added to the graph")
+	}
+
+	inputSlice1 := []int32{1, 2, 3}
+	inputSlice2 := []int32{3, 4, 5}
+
+	t1, err := tf.NewTensor(inputSlice1)
+	if err != nil {
+		t.Error("Problem trying create a new tensor, Error:", err)
```

Comment:
This and later uses of t.Error("..."), t.FailNow()
can all be replaced with t.Fatal("...")
(Fatal is the equivalent of Log + FailNow())